REINDEXING.md

```diff
@@ -0,0 +1,261 @@
+merger                 ECR tag updated,
+palette_inferrer       ECR tag updated,
+relation_embedder      ECR tag updated,
+router                 ECR tag updated,
+transformer_calm       ECR tag updated,
+transformer_mets       ECR tag updated,
+transformer_miro       ECR tag updated,
+transformer_sierra     ECR tag updated,
+transformer_tei        ECR tag updated,
+
+Deployed release eeb9482f-0a6a-4211-8de9-27cee7567134 to EXAMPLE-PIPELINE (Environment(id='EXAMPLE-PIPELINE', name='EXAMPLE-PIPELINE'))
+
+Checking deployment of eeb9482f-0a6a-4211-8de9-27cee7567134 to EXAMPLE-PIPELINE
+Allowing 600s for deployment.
+
+Deployment of eeb9482f-0a6a-4211-8de9-27cee7567134 to EXAMPLE-PIPELINE successful
+Deployment took 2s
+```
+
+This operation **only tags ECR images with the new environment** it does not deploy any services. 
+
+This will be done when you terraform a new pipeline.
+
+### Terraform a new pipeline
+
+You will now need to create a new pipeline module in [./pipeline/terraform/main.tf](./pipeline/terraform/main.tf).
+
+Copy and paste an exisiting pipeline making sure to update the fields:
+
+- `pipeline_date`: References secrets required to access ES and to sets internal infrastructure labels.
+- `release_label`: Sets the ECR label to use on the service deployment images, created in the above deployment process. 
+- `is_reindexing`: Sets ES cluster/service scaling limits while reindexing, and connects reindexing topics. Enabling this will incur above normal costs for a pipeline.
+
+See the following example:
+
+```tf
+module "catalogue_pipeline_2021-07-06" {
+  source = "./stack"
+
+  pipeline_date = "2021-07-06"
+  release_label = "2021-07-06"
+
+  is_reindexing = false
+
+  # Boilerplate that shouldn't change between pipelines.
+  # ...
+}
+
+module "catalogue_pipeline_MY-NEW-PIPELINE" {
+  source = "./stack"
+
+  pipeline_date = "MY-NEW-PIPELINE"
+  release_label = "MY-NEW-PIPELINE"
+
+  is_reindexing = true
+
+  # Boilerplate that shouldn't change between pipelines.
+  # ...
+}
+```
+
+### Running the reindex script
+
+Now we have our pipeline connected for reindexing and running our chosen version of the pipeline code we can start a reindex operation.
+
+The reindex script can be found in [./reindexer/start_reindex.py](./reindexer/start_reindex.py).
+
+```
+> python3 ./start_reindex.py
+
+Which source do you want to reindex? (all, miro, sierra, mets, calm): all
+Which pipeline are you sending this to? (catalogue, catalogue_miro_updates, reporting): catalogue
+Every record (complete), just a few (partial), or specific records (specific)? (complete, partial, specific): partial
+```
+
+You may wish to run a partial-reindex to verify that the process will work as expected before embarking on a complete reindex.
+
+You can monitor a reindex in progress using Grafana at [https://monitoring.wellcomecollection.org/](https://monitoring.wellcomecollection.org/), or by looking at CloudWatch metrics in the `platform` AWS account.
```

Comment:
We also have the `get_reindex_status.py` script: https://github.com/wellcomecollection/catalogue-pipeline/blob/main/reindexer/get_reindex_status.py which reports:

- the total number of source records
- how many works have reached each stage of the reindex